src/runtime.jl

```diff
@@ -125,79 +122,17 @@ Get the current task ID for this worker in hex format.
 """
 get_task_id() = String(ray_jll.Hex(ray_jll.GetCurrentTaskId(ray_jll.GetCoreWorker())))
 
-function parse_ray_args_from_raylet_out(session_dir)
-    #=
-    "Starting agent process with command: ... \
-    --node-ip-address=127.0.0.1 --metrics-export-port=60404 --dashboard-agent-port=60493 \
-    --listen-port=52365 --node-manager-port=58888 \
-    --object-store-name=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/sockets/plasma_store \
-    --raylet-name=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/sockets/raylet \
-    --temp-dir=/tmp/ray --session-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385 \
-    --runtime-env-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/runtime_resources \
-    --log-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/logs \
-    --logging-rotate-bytes=536870912 --logging-rotate-backup-count=5 \
-    --session-name=session_2023-08-14_14-54-36_055139_41385 \
-    --gcs-address=127.0.0.1:6379 --minimal --agent-id 470211272"
-    =#
-    line = open(joinpath(session_dir, "logs", "raylet.out")) do io
-        while !eof(io)
-            line = readline(io)
-            if contains(line, "Starting agent process")
-                return line
-            end
-        end
-    end
-
-    line !== nothing || error("Unable to locate agent process information")
-
-    # --raylet-name=/tmp/ray/session_2023-08-14_18-52-23_003681_54068/sockets/raylet
-    raylet_match = match(r"raylet-name=((\/[a-z,0-9,_,-]+)+)", line)
-    raylet = if raylet_match !== nothing
-        String(raylet_match[1])
-    else
-        error("Unable to find Raylet socket")
-    end
-
-    # --object-store-name=/tmp/ray/session_2023-08-14_18-52-23_003681_54068/sockets/plasma_store
-    store_match = match(r"object-store-name=((\/[a-z,0-9,_,-]+)+)", line)
-    store = if store_match !== nothing
-        String(store_match[1])
-    else
-        error("Unable to find Object Store socket")
-    end
+function get_node_to_connect_for_driver(global_state_accessor, node_ip_address)
+    node_to_connect = StdString()
+    status = ray_jll.GetNodeToConnectForDriver(global_state_accessor, node_ip_address,
+                                               CxxPtr(node_to_connect))
```

Comment:
it's probably a good idea to check here whether the status is good; if the GCS can't be reached after a timeout (something like 60s), it'll terminate the whole process, and that failure can't be detected when calling `Connect` (it tries to connect in a separate thread)

Reply:
I guess you'd fail to like parse/destructure the node info though if it the call returned junk